src/backend/sev/builder.rs

```diff
@@ -9,34 +9,71 @@ use crate::backend::kvm::builder::kvm_try_from_builder;
 use crate::backend::kvm::mem::Region;
 
 use std::convert::TryFrom;
+use std::fmt::Debug;
 use std::sync::{Arc, RwLock};
+use std::{thread, time};
 
 use anyhow::{Context, Error, Result};
 use kvm_ioctls::Kvm;
 use mmarinus::{perms, Map};
 use primordial::Page;
+use rand::{thread_rng, Rng};
 use sallyport::elf::pf::snp::{CPUID, SECRETS};
 use x86_64::VirtAddr;
 
+const SEV_RETRIES: usize = 3;
+const SEV_RETRY_SLEEP_MS: u64 = 500;
+
 pub struct Builder {
     kvm_fd: Kvm,
     launcher: Launcher<Started, Firmware>,
     regions: Vec<Region>,
     sallyports: Vec<Option<VirtAddr>>,
 }
 
+fn retry<O, E, F>(func: F) -> Result<O, E>
+where
+    F: Fn() -> Result<O, E>,
+    E: Debug,
+{
+    let mut retries = SEV_RETRIES;
+    let mut rng = thread_rng();
+    // try to open /dev/sev several times
+    loop {
+        match func() {
+            Err(e) => {
+                if retries > 0 {
+                    retries -= 1;
+                    eprintln!("Error {:#?}.\nRetrying soon.", e);
+                    let millis =
+                        time::Duration::from_millis(SEV_RETRY_SLEEP_MS + rng.gen::<u8>() as u64);
+                    thread::sleep(millis);
+                    continue;
+                }
+                return Err(e);
```

Comment:
nit; We could maybe get rid of the nested `if` by adding a guard? e.g. something like this:
```suggestion
            Err(e) if retries == 0 => {
                return Err(e);
            }
            Err(e) if retries > 0 {
                retries -= 1;
                eprintln!("Error {:#?}.\nRetrying soon.", e);
                let millis =
                    time::Duration::from_millis(SEV_RETRY_SLEEP_MS + rng.gen::<u8>() as u64);
                thread::sleep(millis);
                continue;
            }
```

Would be nice to log the retry number as well, maybe?

Reply:
addressed